src/app_service/service/wallet_account/service.nim

```diff
@@ -207,128 +217,69 @@ method getWalletAccount*(self: Service, accountIndex: int): WalletAccountDto =
 method getCurrencyBalance*(self: Service): float64 =
   return self.getWalletAccounts().map(a => a.getCurrencyBalance()).foldl(a + b, 0.0)
 
-method getDefaultAccount(self: Service): string =
-  return status_go_eth.getAccounts().result[0].getStr
-
-method saveAccount(
-  self: Service,
-  address: string,
-  name: string,
-  password: string,
-  color: string,
-  accountType: string,
-  isADerivedAccount = true,
-  walletIndex: int = 0,
-  id: string = "",
-  publicKey: string = "",
-): string =
-  try:
-    status_go_accounts.saveAccount(
-      address,
-      name,
-      password,
-      color,
-      accountType,
-      isADerivedAccount = true,
-      walletIndex,
-      id,
-      publicKey,
-    )
-    let accounts = fetchAccounts()
-    let prices = self.fetchPrices()
+method addNewAccountToLocalStore(self: Service) =
+  let accounts = fetchAccounts()
+  let prices = self.fetchPrices()
 
-    var newAccount = accounts[0]
-    for account in accounts:
-      if not self.accounts.haskey(account.address):
-        newAccount = account
-        break
+  var newAccount = accounts[0]
+  for account in accounts:
+    if not self.accounts.haskey(account.address):
+      newAccount = account
+      break
 
-    newAccount.tokens = self.buildTokens(newAccount, prices)
-    self.accounts[newAccount.address] = newAccount
-    self.events.emit("walletAccount/accountSaved", AccountSaved(account: newAccount))
-  except Exception as e:
-    return fmt"Error adding new account: {e.msg}"
+  let balances = self.fetchBalances(@[newAccount.address])
+  newAccount.tokens = self.buildTokens(newAccount, prices, balances{newAccount.address})
+  self.accounts[newAccount.address] = newAccount
+  self.events.emit("walletAccount/accountSaved", AccountSaved(account: newAccount))
```

Comment:
You can put the name in a const

Reply:
@jrainville true, but that's not my code! :) I just took the above mentioned commit and apply it in the `master` branch,  so just was moving the code around without looking into the details of each proc particularly.

Anyway, I added new commit where signal names are set as string constants and used across the app that way.